Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/presencetracing/ui/poster/QrCodePosterTestViewModel.kt

```diff
@@ -84,13 +89,21 @@ class QrCodePosterTestViewModel @AssistedInject constructor(
             isRunning = true
             val traceLocation = traceLocation()
             val correctionLevel = appConfigProvider.getAppConfig().presenceTracing.qrCodeErrorCorrectionLevel
-            val qrCode = qrCodeGenerator.createQrCode(
-                input = traceLocation.locationUrl,
-                length = length,
-                margin = 0,
-                correctionLevel = correctionLevel
-            )
-            qrCodeBitmap.postValue(qrCode)
+
+            val drawable = run {
+                val req = ImageRequest.Builder(context).apply {
+                    data(
+                        CoilQrCode(
+                            content = traceLocation.locationUrl,
+                            options = QrCodeOptions(correctionLevel = correctionLevel)
+                        )
+                    )
+                    size(length)
```

Comment:
size could be also specified in the `options`

Reply:
The options are qrcode **data**  specific (like error correction level and charset), while the size is a UI detail.
The options go towards the data generation part (data matrix) and do not change depending on the size of the target.

* Step 1 generate data (doesn't need size)
* Step 2 draw data into bitmap (can re-use the data from step 1 for different bitmap sizes)

If we introduce the size as part of the options, the generated data matrix could no longer be re-used for images of different sizes as the `QrCodeOptions` would be different.

We only need to specify the size at all here, because the way the poster generation currently works, we don't have a target to give to `Coil`, so the library can't figure out the size itself.